src/test_utils.rs

```diff
@@ -112,6 +113,18 @@ pub mod tests {
         annotations
     }
 
+    pub fn parse_and_validate_buffered(src: &str) -> String {
```

Comment:
I find this method's name strange. What do you think about, not offering a complete new `parse_and_...` but instead offer a `get_terminal_output(diagnostics: Vec<Diagnostic>)`, so we can test the diagnostics this way, no matter where we got them from.

Reply:
Sounds good, but unless I'm missing something this isn't possible with the given function signature right? Specifically the `get_terminal_output(diagnostics: Vec<Diagnostic>)` needs to create a new `Diagnostician` and register content / a file with it in order to get a correct report. The `Diagnostic` enum has no access to these files however, hence that signature has to be expanded to also take in a `source` parameter. For example
```
#[test]
fn demo() {
	let content = "...";
	let diagnostics = parse_and_validate(content);
	assert_snapshot!(get_terminal_output(content, diagnostics));
}
```
That's possible but will require changes in many (all?) unit tests mainly because `content` isn't declared as a variable but rather fed directly as a string literal argument. Do we still proceed with this?

Reply:
hmm damn you're right. ok lets stick with the current approach for now